File: Juan/P4_FULL/ruby5/Napakalaki/lib/DeathBadConsequence.rb
require_relative 'BadConsequence.rb'
require_relative 'Monster.rb'
require_relative 'Prize.rb'
require_relative 'TreasureKind.rb'
require_relative 'NumericBadConsequence'

module NapakalakiGame

class DeathBadConsequence < NumericBadConsequence 
    
    def initialize(text)
        super(text,0,0,0)
        @death = true
    end
    
    attr_reader :death
    
end
end

File: Juan/P5/java/Napakalaki/src/NapakalakiGame/DeathBadConsequence.java

package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author juane
 */
public class DeathBadConsequence extends NumericBadConsequence{
    
   
 
    private boolean muerte  = false;
    public DeathBadConsequence(String text, boolean death) {
       super( text, Player.MAXLEVEL, BadConsequence.MAXTREASURES, BadConsequence.MAXTREASURES);
       super.levels=1;

       muerte = death;
        
    }
    @Override
    public BadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h){      
       
        super.setnHiddenTreasures(BadConsequence.MAXTREASURES);
        super.setnHiddenTreasures(BadConsequence.MAXTREASURES);
        //super.adjustToFitTreasureList(v, h);
        BadConsequence badconsequence = new DeathBadConsequence(text,true);
        return badconsequence; 
    }
    
    
    
    
}
